packages/fleather/lib/src/widgets/text_selection.dart

```diff
@@ -660,176 +879,777 @@ class EditorTextSelectionGestureDetectorBuilder {
   ///
   /// See also:
   ///
-  ///  * [EditorTextSelectionGestureDetector.onForcePressEnd], which triggers this
+  ///  * [TextSelectionGestureDetector.onForcePressEnd], which triggers this
   ///    callback.
   @protected
   void onForcePressEnd(ForcePressDetails details) {
     assert(delegate.forcePressEnabled);
-    renderEditor!.selectWordsInRange(
+    renderEditor.selectWordsInRange(
       from: details.globalPosition,
       cause: SelectionChangedCause.forcePress,
     );
-    if (shouldShowSelectionToolbar) editor!.showToolbar();
+    if (shouldShowSelectionToolbar) {
+      editor.showToolbar();
+    }
   }
 
-  /// Handler for [EditorTextSelectionGestureDetector.onSingleTapUp].
+  /// Handler for [TextSelectionGestureDetector.onSingleTapUp].
   ///
   /// By default, it selects word edge if selection is enabled.
   ///
   /// See also:
   ///
-  ///  * [EditorTextSelectionGestureDetector.onSingleTapUp], which triggers
+  ///  * [TextSelectionGestureDetector.onSingleTapUp], which triggers
   ///    this callback.
   @protected
-  void onSingleTapUp(TapUpDetails details) {
+  void onSingleTapUp(TapDragUpDetails details) {
```

Comment:
What is the rational behing the change of details type?

Reply:
Before, we were using our own gesture detector but now with `TextSelectionGestureDetector` we have to provide  `GestureTapDragUpCallback` which is a `void Function(TapDragUpDetails details)`. So it should match its signature.